src/cryptonote_protocol/cryptonote_protocol_handler.inl

```diff
@@ -1630,10 +1630,10 @@ skip:
     }
 
     uint64_t n_use_blocks = m_core.prevalidate_block_hashes(arg.start_height, arg.m_block_ids);
-    if (n_use_blocks == 0)
+    if (n_use_blocks <= arg.m_block_ids.size() - HASH_OF_HASHES_STEP)
```

Comment:
Shouldn't it test `if (n_use_blocks + HASH_OF_HASHES_STEP <= arg.m_block_ids.size())` instead in case `arg.m_block_ids.size()` is smaller than `HASH_OF_HASHES_STEP`?